main.tf

```diff
@@ -196,23 +197,27 @@ resource "aws_default_route_table" "default" {
 ################################################################################
 
 resource "aws_route_table" "public" {
-  count = var.create_vpc && length(var.public_subnets) > 0 ? 1 : 0
+  count = var.create_vpc && length(var.public_subnets) > 0 && length(var.firewall_sync_states) == 0 ? 1 : length(var.firewall_sync_states) > 0 ? local.nat_gateway_count : 0
```

Comment:
Why does the number of public routes depend on the number of NAT Gateways?

Reply:
hmm this works well when `one_nat_gateway_per_az=true` because each nat gateway will be in its own protected subnet, and for each zone in the variable `az` so if there user only provides `['1a','1b']` and the region actually has a `1c` region, there will only be 2 NAT gateways, and implies there should be 2 firewall subnets supplied, and 2 public(1a and 1b) route tables, to map to each firewall endpoint. 

Basically i'm reusing the calculation for nat_gateway_count as i think it covers the scenarios best, but i'm not sure what kind of vpc designs people do with this module. 

Could there be a case where there are >1 nat gateway in a availability zone?

Reply:
Ah interesting. From the [AWS Network Firewall Deployment Models](https://aws.amazon.com/blogs/networking-and-content-delivery/deployment-models-for-aws-network-firewall/) model 2 that supports NAT Gateways, it appears that the Private subnets need to route to the NAT, and the Public/Protected subnets need to point to the Network Firewall's VPC Endpoint. [Check out their diagram here](https://d2908q01vomqb2.cloudfront.net/5b384ce32d8cdef02bc3a139d4cac0a22bb029e8/2020/11/18/anfw-public-igw-deployment-high-res1-1-1.png). So in my commit, I changed this to be [one route table per public subnet](https://github.com/terraform-aws-modules/terraform-aws-vpc/commit/c3eca3e0434177d4743c2ee91d364b2193c24989#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR219), so that we could route it to the associated Network Firewall VPC endpoint.